drivers/dai/intel/ssp/ssp.c

```diff
@@ -2468,6 +2418,54 @@ static const struct dai_properties *dai_ssp_get_properties(const struct device *
 	return prop;
 }
 
+static void ssp_acquire_ip(struct dai_intel_ssp *dp)
+{
+	struct dai_intel_ssp_plat_data *ssp = dai_get_plat_data(dp);
+
+	ssp->acquire_count++;
+
+	if (ssp->acquire_count == 1) {
+		/* Enable SSP power */
+		dai_ssp_pm_runtime_en_ssp_power(dp, ssp->ssp_index);
+
+		/* Disable dynamic clock gating before touching any register */
+		dai_ssp_pm_runtime_dis_ssp_clk_gating(dp, ssp->ssp_index);
+	}
+}
+
+static void ssp_release_ip(struct dai_intel_ssp *dp)
+{
+	struct dai_intel_ssp_plat_data *ssp = dai_get_plat_data(dp);
+
+	if (ssp->acquire_count == 0) {
```

Comment:
Can this ever happen in a legal scenario? I..e. is it normal to have unbalanced acquire/release calls?

Reply:
It shouldn't happen in normal/legal scenario. Acquire / release calls should be finally balanced.